core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -671,33 +698,43 @@ private[zio] trait Schedule_Functions extends Serializable {
    * </pre>
    */
   final def spaced(interval: Duration): Schedule[Any, Int] =
-    forever.delayed(_ + interval)
+    forever.delayed(_ + interval.relative)
 
   /**
    * A schedule that always recurs, increasing delays by summing the
    * preceding two delays (similar to the fibonacci sequence). Returns the
    * current duration between recurrences.
    */
-  final def fibonacci(one: Duration): Schedule[Any, Duration] =
-    delayed(unfold[(Duration, Duration)]((Duration.Zero, one)) {
-      case (a1, a2) => (a2, a1 + a2)
-    }.map(_._1))
+  final def fibonacci(one: Duration): Schedule[Any, Duration] = {
+    val f = delayed[Any, Any](
+      unfold[(Delay, Delay)]((Delay.none, one.relative)) {
+        case (a1, a2) => (a2, a1 + a2)
+      }.map(_._1)
+    )
+
+    f.mapM(_.run) // TODO: Dotty doesn't infer this properly
+  }
 
   /**
    * A schedule that always recurs, but will repeat on a linear time
    * interval, given by `base * n` where `n` is the number of
    * repetitions so far. Returns the current duration between recurrences.
    */
-  final def linear(base: Duration): Schedule[Any, Duration] =
-    delayed(forever.map(i => base * i.doubleValue()))
+  final def linear(base: Duration): Schedule[Any, Duration] = {
+    val l = delayed[Any, Any](forever.map(i => (base * i.doubleValue()).relative))
+    l.mapM(_.run) // TODO: Dotty doesn't infer this properly
```

Comment:
This pattern is so common, because of the type inference problems, I suggest that we use `l >>> duration` (with the above definition of `duration` that I suggested), both here and elsewhere. Bonus: Dotty may like it more than `mapM`, so we can eliminate the temporary variables and manual type annotations.

Reply:
Just tried this approach an even with `l >>> duration` dotty complained about missing implicit for `ConformsR`. With manual type annotations things went fine